modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ITIgniteNodeRestartTest.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /**
+     * Gets current network port.
+     *
+     * @param nodeName Node name.
+     * @return Network port.
+     * @throws Exception If some issue has happened when getting the port.
+     */
+    private int getNodePort(String nodeName) throws Exception {
+        Map<String, List<IgniteComponent>> nodesStartedComponents = (Map<String, List<IgniteComponent>>)ReflectionUtils.tryToReadFieldValue(IgnitionImpl.class, "nodesStartedComponents", null).get();
+
+        assertNotNull(nodesStartedComponents.get(nodeName));
+
+        MetaStorageManager metaStorageManager = (MetaStorageManager)nodesStartedComponents.get(NODE_NAME).stream().filter(component -> component instanceof MetaStorageManager).findFirst().get();
+
+        ConfigurationManager locCfgMgr = (ConfigurationManager)ReflectionUtils.tryToReadFieldValue(MetaStorageManager.class, "locCfgMgr", metaStorageManager).get();
+
+        return locCfgMgr.configurationRegistry().getConfiguration(NetworkConfiguration.KEY).port().value();
+    }
+
+    /**
+     * Restarts the node which stores some data.
+     */
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-15255")
+    public void nodeWithDataTest() {
+        Ignite ignite = IgnitionManager.start(NODE_NAME, "{\n" +
+            "  \"node\": {\n" +
+            "    \"metastorageNodes\":[ " + NODE_NAME + " ]\n" +
+            "  },\n" +
+            "  \"network\": {\n" +
+            "    \"port\":3344,\n" +
+            "    \"netClusterNodes\":[ \"localhost:3344\" ]\n" +
+            "  }\n" +
+            "}", workDir.resolve(NODE_NAME));
+
+        SchemaTable scmTbl1 = SchemaBuilders.tableBuilder("PUBLIC", TABLE_NAME).columns(
+            SchemaBuilders.column("id", ColumnType.INT32).asNonNull().build(),
+            SchemaBuilders.column("name", ColumnType.string()).asNullable().build()
+        ).withIndex(
+            SchemaBuilders.pkIndex()
+                .addIndexColumn("id").done()
+                .build()
+        ).build();
+
+        Table table = ignite.tables().getOrCreateTable(scmTbl1.canonicalName(), tbl -> SchemaConfigurationConverter.convert(scmTbl1, tbl)
+            .changePartitions(10));
+
+        for (int i = 0; i < 100; i++) {
+            Tuple key = table.tupleBuilder()
+                .set("id", i)
+                .build();
+
+            Tuple vaul = table.tupleBuilder()
+                .set("name", "name " + i)
+                .build();
+
+            table.kvView().put(key, vaul);
+        }
+
+        IgnitionManager.stop(NODE_NAME);
+
+        ignite = IgnitionManager.start(NODE_NAME, null, workDir.resolve(NODE_NAME));
+
+        assertNotNull(ignite.tables().table(TABLE_NAME));
+
+        for (int i = 0; i < 100; i++) {
+            assertEquals("name " + i, table.kvView().get(table.tupleBuilder()
+                .set("id", i)
+                .build())
+                .stringValue("name"));
+        }
+
+        IgnitionManager.stop(NODE_NAME);
+    }
+}
```

Comment:
Let's also add a test that will run node with explicit configuration with two properties and later restart the node with new startup configuration that will override only one property. After restart let's check that overridden property was changed and not-overridden one didn't

Reply:
Added a new test (twoCustomPropertiesTest).